types/uuid.go

```diff
@@ -51,14 +51,14 @@ func (b *UUIDBuilder) AppendValues(v []uuid.UUID, valid []bool) {
 		panic("len(v) != len(valid) && len(valid) != 0")
 	}
 
-	data := make([][]byte, len(v))
+	data := make([]string, len(v))
 	for i := range v {
 		if len(valid) > 0 && !valid[i] {
 			continue
 		}
-		data[i] = v[i][:]
+		data[i] = v[i].String()
 	}
-	b.ExtensionBuilder.Builder.(*array.FixedSizeBinaryBuilder).AppendValues(data, valid)
```

Comment:
I do think fixed size binary is more suitable here, per the fact that we know that the len will match uuid.UUID len and we don't need to make the check for that

Reply:
I agree, also parquet has built-in uuid logical-type which is based on fixed size binary: https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#uuid